skxray/core/roi.py

```diff
@@ -357,139 +355,87 @@ def roi_pixel_values(image, labels, index=None):
     for n in index:
         roi_pix.append(image[labels == n])
     return roi_pix, index
+    
 
-
-def mean_intensity_sets(images_set, labels):
-    """
-    Mean intensities for ROIS' of the labeled array for different image sets
+def mean_intensity(images, labeled_array, index=None):
+    """Compute the mean intensity for each ROI in the image list
 
     Parameters
     ----------
-    images_set : array
-        images sets
-        shapes is: (len(images_sets), )
-        one images_set is iterable of 2D arrays dimensions are: (rr, cc)
-
-    labels : array
+    images : list
+        List of images
+    labeled_array : array
         labeled array; 0 is background.
-        Each ROI is represented by a distinct label (i.e., integer).
+        Each ROI is represented by a nonzero integer. It is not required that
+        the ROI labels are contiguous
+    index : int, list, optional
+        The ROI's to use. If None, this function will extract averages for all
+        ROIs
 
     Returns
     -------
-    mean_intensity_list : list
-        average intensity of each ROI as a list
-        shape len(images_sets)
-
-    index_list : list
-        labels list for each image set
-
-    """
-    return tuple(map(list,
-                     zip(*[mean_intensity(im,
-                                          labels) for im in images_set])))
-
-
-def mean_intensity(images, labels, index=None):
-    """
-    Mean intensities for ROIS' of the labeled array for set of images
-
-    Parameters
-    ----------
-    images : array
-        Intensity array of the images
-        dimensions are: (num_img, num_rows, num_cols)
-
-    labels : array
-        labeled array; 0 is background.
-        Each ROI is represented by a distinct label (i.e., integer).
-
+    mean_intensity : list
```

Comment:
Yeah, I'm with Tom on this one. Can you undo the change to the code and update the docs to say that it is coming back as an array?